modules/puppet/templates/govuk_puppet

```diff
@@ -8,11 +8,223 @@ fi
 
 START_TIME=`date +%s%3N`
 
-# Allow return code 2 in case -v/--detailed-exitcodes is passed.
-RBENV_VERSION=1.9.3 /usr/local/bin/govuk_setenv default puppet agent --onetime --no-daemonize "$@" || [ $? -eq 2 ]
+remove_all_locks() {
 
-TIME_TOOK=$((`date +%s%3N`-START_TIME))
-echo "<%= @fqdn_metrics %>.puppet.run_duration:${TIME_TOOK}|ms" | nc -w 1 -u localhost 8125
+	if ls *.lock 1> /dev/null 2>&1; then #there are locks
 
-# We should only reach here if a last_run_summary was written.
-/usr/local/bin/puppet_passive_check_update >/dev/null
+                        for filename in *.lock
+                        do
+                                rm $filename
+                                echo "Removed: $filename"
+                        done
+                        do_puppet
+
+                else #there are no locks left, it should be enabled already, but just in case
+
+                        do_puppet
+
+                fi
+
+}
+
+listlocks() {
+
+	#list all current locks, date, name, reason, identifier
+        #simplify this structure. push the date and a (kind of) 
+	#uuid to the filename. Keep the reason, name and uuid as
+	#the lockfile content
+
+	#NB The last field (the lock identifier) can be grabbed with
+	#a simple awk '{print $NF}' in a script
+
+	if ls *.lock 1> /dev/null 2>&1; then
+
+        	for filename in *.lock
+               	 	do
+               	        	line=$(head -n 1 "$filename") #Grab file contents
+               	         	echo "$filename : $line"
+               	 	done
+
+	else
+		echo "No lock files present"
+	fi
+
+}
+
+do_puppet() {
+
+	# Allow return code 2 in case -v/--detailed-exitcodes is passed.
+	RBENV_VERSION=1.9.3 /usr/local/bin/govuk_setenv default puppet agent --onetime --no-daemonize "$@" || [ $? -eq 2 ]
+
+	TIME_TOOK=$((`date +%s%3N`-START_TIME))
+	echo "<%= @fqdn_metrics %>.puppet.run_duration:${TIME_TOOK}|ms" | nc -w 1 -u localhost 8125
+
+	# We should only reach here if a last_run_summary was written.
+	/usr/local/bin/puppet_passive_check_update >/dev/null
+	exit 0;
+
+}
+
+
+usage() {
+
+	programname=$0
+
+	echo "usage: $programname"
+	echo "	--list-locks : show all current locks "
+	echo "	--remove-lock : remove one lock, takes numeric identifier as parameter "
+	echo "	--disable : add a lock "
+	echo "	--enable : remove all locks "
+
+	exit 1
+
+}
+
+PUPPET_STATE_DIR=`puppet config print statedir`
```

Comment:
Is this user specific? Might be easier just to have it in `/var/run/puppet/` or something